### PR TITLE
Feature/gh 631 update to angular 11

### LINKED ISSUE #631: Update SPA to Angular 11

### PR DESCRIPTION
**Pull request by netzartist**
Followup of https://github.com/opendevstack/ods-provisioning-app/pull/629#issuecomment-725923282, closes #631

### CODE REVIEW

File: client/prettier.config.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  printWidth: 80,
+  printWidth: 140,
```
Comment (cschweikert) [inline client/prettier.config.js:2]: This little change in the prettier rules actually triggers a lot of changes from code formatting. From my point of view this reformatting is not needed for the actual update to Angular 11. It is hard for me to see, what changes are acually related to the Angular 11 update itself. Maybe the reformatting can be moved into a separate PR.

Comment (netzartist) [reply client/prettier.config.js:2]: I grouped the changes into several [commits](https://github.com/opendevstack/ods-provisioning-app/pull/632/commits), maybe it's easier to follow them that way?

Comment (cschweikert) [reply client/prettier.config.js:2]: True true. Just was surprised by the amount of changed files 😜

**Review by cschweikert — DISMISSED** [review]

File: client/package.json [hunk 2]
```diff
@@ -11,42 +11,54 @@
...
+    "@angular/compiler-cli": "~11.0.0",
+    "@angular/language-service": "~11.0.0",
     "@ngneat/spectator": "^5.3.1",
     "@types/jest": "25.1.4",
     "@types/node": "^12.11.1",
+    "@typescript-eslint/eslint-plugin": "4.3.0",
+    "@typescript-eslint/parser": "4.3.0",
     "bootstrap": "^4.4.1",
     "codelyzer": "^5.1.2",
+    "eslint": "^7.6.0",
```
Comment (cschweikert) [inline client/package.json:54]: Can the ``tslint``-related dependencies below be removed?

File: client/prettier.config.js (hunk 1 above)
Comment (netzartist) [reply client/prettier.config.js:2]: Totally understandable :-) But you were right, lint changes are not part of the update to 11 itself and would have been easier to follow in a separate PR.

File: client/package.json (hunk 2 above)
Comment (netzartist) [reply client/package.json:54]: Yes, thanks for the hint! Also, `codelyzer` and `tslint-config-prettier` are not needed anymore.

**Review by cschweikert — APPROVED** [review]
